### PR TITLE
Estimate fee for `DEPLOY_ACCOUNT` transactions

### LINKED ISSUE #426: Support estimating fee for `DEPLOY_ACCOUNT` transactions

### PR DESCRIPTION
**Pull request by cptartur**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #426 

* **What is the current behavior?** (You can also link to an open issue here)

### CODE REVIEW

File: starknet_py/tests/e2e/client/fixtures/transactions.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    new_gateway_account_client: AccountClient,
+) -> DeployAccount:
+    account_client = new_gateway_account_client
+
+    compiled_contract = Path(
+        contracts_dir / "contract_with_validate_deploy.cairo"
+    ).read_text("utf-8")
+    declare_tx = await account_client.sign_declare_transaction(
+        compilation_source=compiled_contract, max_fee=MAX_FEE
+    )
```
Comment (war-in) [inline starknet_py/tests/e2e/client/fixtures/transactions.py:21]: Are you sure we should name this variable `compiled_contract`? It is not compiled 😕

File: starknet_py/tests/e2e/client/fixtures/transactions.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    declare_result = await account_client.declare(declare_tx)
+    await account_client.wait_for_tx(declare_result.transaction_hash)
+
+    return await new_gateway_account_client.sign_deploy_account_transaction(
+        class_hash=declare_result.class_hash,
+        contract_address_salt=10,
+        constructor_calldata=[],
+        max_fee=10000000000,
```
Comment (war-in) [inline starknet_py/tests/e2e/client/fixtures/transactions.py:29]: Maybe use `MAX_FEE`?

File: starknet_py/tests/e2e/conftest.py [hunk 3]
```diff
@@ -53,11 +53,16 @@
...
+MAX_FEE = int(1e20)
+
 mock_dir = Path(os.path.dirname(__file__)) / "mock"
 typed_data_dir = mock_dir / "typed_data"
 contracts_dir = mock_dir / "contracts"
 
 
+# This is needed for importing fixtures from `fixtures` directory
+pytest_plugins = ["starknet_py.tests.e2e.client.fixtures.transactions"]
+
```
Comment (war-in) [inline starknet_py/tests/e2e/conftest.py:65]: Nice! Could you please create an issue about moving all of the fixtures to separate files?

Comment (cptartur) [reply starknet_py/tests/e2e/conftest.py:65]: See #441

**Pull request merged**